src/mds/Migrator.cc

```diff
@@ -1483,7 +1483,7 @@ void Migrator::handle_export_prep_ack(const cref_t<MExportDirPrepAck> &m)
 
   if (!m->is_success()) {
     dout(7) << "peer couldn't acquire all needed locks or wasn't active, canceling" << dendl;
-    export_try_cancel(dir, false);
+    export_try_cancel(dir);
```

Comment:
I don't think this is needed. In this export_prepping state, bystander mds aren't aware of there is subtree export. So cancel export does not need to notify bystanders.

Reply:
Thanks for your reply @ukernel.  In export_prepping state, cancel export does not need to notify bystanders.  I have some doubts, in this state, does peer mds need to execute cancel export?  Because peer mds has added the import task to import_state and the import task is in IMPORT_PREPPING state.

Reply:
no, the cancel is initiated by importer. importer already cleanup its states